### ISSUE TITLE
关于generate生成的结果的问题

### ISSUE THREAD

**Issue by Tian14267** [Facico/Chinese-Vicuna#71]
大神们好。我在`generate`的时候，遇到生成的结果，并不对。比如：
**输入：** `自然语言处理是什么`
**输出：** `自然语言处理是什么？自然语言处理是什么？自然语言处理是什么？自然语言处理是什么？自然语言处理是什么？自然语言处理是什么？自然语言处理是什么？自然语言处理是什么？自然语言处理是什么？自然语言处理是什么？自然语言处理是`

这个明显不太对。请问是啥情况啊。我试了原始的预训练模型，和下载的`finetune`模型，都是这个结果。下面这个压缩包是我`generate`的代码。
[generate_fffan.zip](https://github.com/Facico/Chinese-Vicuna/files/11220203/generate_fffan.zip)

**Comment by Facico** [comment]
你这个程序我跑也有问题，你试过我们给出的推理脚本吗，应该是没有问题的

**Comment by Facico** [comment]
你可以试试[这里面](https://github.com/Facico/Chinese-Vicuna/blob/master/docs/problems.md)问题3的生成代码，然后把对应参数和prompt按照我们generate.py里面加上去就可以了

**Comment by Tian14267** [comment]
@Facico 大神，我使用问题3里面那个代码，出来的结果是这个
![image](https://user-images.githubusercontent.com/27938135/232468499-d7f81a6f-93c8-46d7-9571-06d564c4b6e5.png)
感觉还是不对。环境也是新建立的，跟你版本对应的。

**Comment by Facico** [comment]
你用的lora文件是什么，可以用我们"Facico/Chinese-Vicuna-lora-7b-3epoch-belle-and-guanaco"这个试试。
同时可以看到，输入开头的第一个token是0，[我们的结果是1](https://github.com/Facico/Chinese-Vicuna/issues/2#issuecomment-1482212014)，怀疑tokenizer的配置也有不同，可以参考这个[issue](https://github.com/Facico/Chinese-Vicuna/issues/59)，修改一下相关的tokenizer配置

**Comment by sevenold** [comment]
我用Belle_open_source_1M数据训练,{'loss': 0.7244, 'learning_rate': 0.00025804870673613693, 'epoch': 4.2}

测试我们自己训练的lora权重
![image](https://user-images.githubusercontent.com/41443157/232987536-24b1f6f1-4b52-4486-a44a-386a70c60fd7.png)


测试了Facico/Chinese-Vicuna-lora-7b-3epoch-belle-and-guanaco权重
![image](https://user-images.githubusercontent.com/41443157/232987433-15339574-dfec-4225-88c8-13b76ab0c62a.png)


请问这个是什么问题？谢谢 @Facico

**Comment by Facico** [comment]
@sevenold 把beams调到>1（这些参数你可以自己多试试）

**Comment by sevenold** [comment]
### Facico/Chinese-Vicuna-lora-7b-3epoch-belle-and-guanaco
![image](https://user-images.githubusercontent.com/41443157/232989022-8a6488ba-f7a9-4e17-ac5d-e596b5178e07.png)
### 测试我们自己训练的lora权重

![image](https://user-images.githubusercontent.com/41443157/232989724-4d25a830-b1fc-40ef-bd1a-2957ae52e0e6.png)

官方提供的好像还是有这个问题，我再多试试参数

**Comment by Facico** [comment]
你这个情况好像是一直没有输出eos，可以参考一下这个[issue](https://github.com/Facico/Chinese-Vicuna/issues/59)

**Comment by Tian14267** [comment]
@Facico @sevenold
这几个参数都是啥意思啊？每太搞明白
![image](https://user-images.githubusercontent.com/27938135/233027489-f40d345c-3a67-40a2-b835-09543d87431e.png)

**Comment by sevenold** [comment]
@Tian14267 可以看看这个https://blog.csdn.net/muyao987/article/details/125917234

**Comment by jinghan23** [comment]
@sevenold 您好！请问您训练的中文lora权重可以开源分享吗？感谢！
